scripts/parse.py

```diff
@@ -515,46 +534,58 @@ def indentandrawoffset2globaloffset(indent: bool, rawoffset: int, offsetlist: Li
     raise RuntimeError('There\'s a bug in this code. This statement should not have been reached.')
 
 
-if __name__ == '__main__':
-    if len(sys.argv) != 2:
-        print('Usage: %s <captureddatahexdump>' % sys.argv[0])
-        print('')
-        print('This program will parse a wireshark hexdump of communication')
-        print('between the TA client and the login server (port 9000) and')
-        print('dump it into a text file')
-        exit(0)
-
-    infilename = sys.argv[1]
-
-    with open(infilename, 'rt') as infile:
-        with open(infilename + '_parsed.txt', 'wt') as outfile:
+@click.command()
```

Comment:
It's not _totally_ necessary but if we are going to have command line flags on parse.py I'd prefer to keep it. Things like the ability to specify additional value annotation files would be less nice without it.

Reply:
I'd rather have that in this development tool than add a dependency for the whole server.

Reply:
Would it be viable to scope dependencies to the script and server folders individually, rather than sharing one `setup.py` between them? I agree dependencies for scripts shouldn't really need to be dragged in for the server, but it'd also be nice to not have to fastidiously avoid all non-standard-lib stuff in the scripts.

Reply:
That would depend on how we package it. One packages has one set of deps. I don't want to create separate packages, but maybe it's possible to have a "dev-only" set of deps.

Reply:
I found this which is one option (see the September 2016 update): https://stackoverflow.com/questions/28509965/setuptools-development-requirements

Reply:
That being said, I'm not sure it is a _huge_ deal if users of the server end up with dependencies for the scripts.

If we're bundling the scripts with the server then surely we'd want them to have the dependencies of both? And if we're not, they should probably be separate packages with separate setup.py (or requirements.txt) files.

Reply:
I have given it some more thought and I want to avoid this dependency.

> Things like the ability to specify additional value annotation files would be less nice without it.

Why not just `parser.add_argument('-f','--file',action='append')` ?

Reply:
Yeah, I'll swap it to use argparse, it's not a huge issue.

Tbh though in general I don't think we should stress too much about adding dependencies for improving the dev experience; realistically there are very few people who are ever going to run the login server and I think we can safely assume they'll be okay with having a few pip packages installed.

If we _really_ need to, there are ways of bundling dependencies in, so we could produce a server binary that doesn't require any dependency management on the user's side.